qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1060,10 +1064,20 @@ def share_level(self, new_level: str) -> None:
                 specified. For example, IBM Quantum experiment service allows
                 "public", "hub", "group", "project", and "private".
         """
-        self._share_level = new_level
+        self._set_share_level_without_save(new_level)
         if self.auto_save:
             self.save_metadata()
 
+    def _set_share_level_without_save(self, new_level: str) -> None:
```

Comment:
Should be unnecessary

Reply:
See https://github.com/Qiskit/qiskit-experiments/pull/364#discussion_r724045468